### PR TITLE
Add ODH support in ODS_CI tests

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Allow ODS_CI to run test-suits on ODH codebase

### CODE REVIEW

**Review by apodhrad — DISMISSED** [review]

File: ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot
Comment (jstourac) [inline ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:88]: Yeah, let's hope that this won't break our automation - did a brief look for our nightly and hopefully shouldn't :crossed_fingers:

File: ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot
Comment (jstourac) [inline ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:218]: Is this used somewhere now?

File: ods_ci/test-variables.yml.example
Comment (jstourac) [inline ods_ci/test-variables.yml.example:80]: Yeah, this is good, but we need to update our automation too to reflect this. Will share in PM.

File: ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot
Comment (vaibhavjainwiz) [reply ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:89]: I had tested it for this code with RHODS and ODH operator by running below cmd :

```
./run_robot_test.sh --include install --test-variable TEST_ENV:CRC --test-variable cluster_type:selfmanaged --test-variable RHODS_OSD_INSTALL_REPO
:https://gitlab.cee.redhat.com/data-hub/olminstall.git --test-variable INSTALL_TYPE:OperatorHub --skip-oclogin true --test-variable UPDATE_CHANNEL:fast --test-case tasks/Tasks/rhods_olm.robot --test-variab
les-file test-variables.yml

```

File: ods_ci/test-variables.yml.example
Comment (vaibhavjainwiz) [reply ods_ci/test-variables.yml.example:80]: Could you please help me to identify which automation file you are refering here. I would verify these changes.

File: ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot
Comment (jstourac) [inline ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:83]: @aloganat this isn't necessary usually at least for our Jenkins jobs. Why we need it here now? Do we do something different in Jenkins? From a brief look to our code there, I don't think so :thinking:

File: ods_ci/test-variables.yml.example
Comment (vaibhavjainwiz) [reply ods_ci/test-variables.yml.example:80]: sure

File: ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot
Comment (vaibhavjainwiz) [reply ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:84]: @aloganat @tarukumar Could you please check it once

Comment (aloganat) [reply ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:84]: @jstourac yes we need to create only DSC for self managed instance. @vaibhavjainwiz Could you test this once without having this line ?

Comment (vaibhavjainwiz) [reply ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:84]: As I check, on `RHODS` DSCI is created automatically when operator installed, where as on ODH, developer need to explicitly create the DSCI resource. I verified, my solution with RHODS and ODH both. Its work for both.






[3:35](https://redhat-internal.slack.com/archives/D06EMGEPC4S/p1712225150535939)
I verified, my solution with RHODS and ODH both. Its work for both.

Comment (vaibhavjainwiz) [reply ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:84]: To avoid any issues in working code, I am adding an check so that new `DSCI` would only be created for ODH.

Comment (vaibhavjainwiz) [reply ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot]: done.. I had pushed the changes. @jstourac Could you please review it again.

**Review by aloganat — COMMENTED** [review]
changes look good to me, but please ensure that the latest changes works well with rhoai nightlies , odh nightlies and ODH.

**Comment by vaibhavjainwiz** [comment]
> changes look good to me, but please ensure that the latest changes works well with rhoai nightlies , odh nightlies and ODH.

I had tested these changes on my local system for RHOAI and ODH operator. How could I verify them for rhoai nightlies , odh nightlies?

File: ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot
Comment (bdattoma) [inline ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot:243]: instead of using sed we have cleaner way: `Create File From Template`. It applies the value of the variables in a given yaml

```yaml
KEY_1: ${value_1}
KEY2: ${value_2}
...
```
```
Set Task Variable    ${value_1}     myval1
Set Task Variable    ${value_2}     myval2
Create File From Template    <template-yaml-path>    <output-yaml-file-path>
```

You can find example by searching for `Create File From Template` in ods-ci.

Anyways, feel free to leave sed commands, work as well

**Review by bdattoma — APPROVED** [review]

File: ods_ci/tasks/Resources/Files/dsci_template.yml
Comment (tarukumar) [inline ods_ci/tasks/Resources/Files/dsci_template.yml:15]: ca bundle part is missing 

`  trustedCABundle:
    managementState: "Managed"
    customCABundle: ""`

Comment (vaibhavjainwiz) [reply ods_ci/tasks/Resources/Files/dsci_template.yml:15]: I had the `trustedCABundle` in DSCInitialization Spec.
```
    trustedCABundle:
        customCABundle: ''
        managementState: Managed
```

**Review by tarukumar — APPROVED** [review]

**Review by bdattoma — APPROVED** [review]
lgtm, pls update the branch with changes from master branch before merging

**Comment by jiridanek** [comment]
@vaibhavjainwiz do you have permissions to merge this yourself? usually it's best when PR author clicks the merge for themselves

**Review by mwaykole — APPROVED** [review]

**Comment by jiridanek** [comment]
Tested in jenkins at https://red.ht/4aGEIzg, feel free to run again with more tests to be better assured this works

**Review by jiridanek — APPROVED** [review]

**Comment by tarukumar** [comment]
@vaibhavjainwiz current code will break ODh-nightly installation. If we want to merge this pr i would suggest to bring back the condition of odh-nightly as earlier. or if we want to use olminstall we still need condition for that  until the operator name is not consistence between odh-nightly and upstream ODH

**Review by tarukumar — APPROVED** [review]

**Pull request merged**
